website/src/components/markdown-layout.js

```diff
@@ -3,37 +3,196 @@ import PropTypes from 'prop-types';
-const H1 = styled.h1``;
-const H2 = styled.h2``;
-const H3 = styled.h3``;
-const H4 = styled.h4``;
-const H5 = styled.h5``;
-const H6 = styled.h6``;
-const ThematicBreak = styled.div``;
-const Blockquote = styled.blockquote``;
-const Ul = styled.ul``;
-const Ol = styled.ol``;
+const TypographyPage = styled.div`
+  font-family: 'Raleway', sans-serif;
+  font-weight: 400;
+  line-height: 1.45;
+  padding: ${customProperties.spacingM};
+`;
+const headerStyles = () => css`
+  line-height: 1.15;
+  margin: 2.75rem 0 1rem;
+`;
+
+// Typography sizes have been calculated from https://type-scale.com
+// Ref: https://medium.com/sketch-app-sources/exploring-responsive-type-scales-cf1da541be54
+const Paragraph = styled.p`
+  margin-bottom: 1.25em;
+`;
+const H1 = styled.h1`
+  ${headerStyles};
+  font-size: 4.209em;
+  margin-top: 0;
+`;
+const H2 = styled.h2`
+  ${headerStyles};
+  font-size: 3.157em;
+`;
+const H3 = styled.h3`
+  ${headerStyles};
+  font-size: 2.369em;
+`;
+const H4 = styled.h4`
+  ${headerStyles};
+  font-size: 1.777em;
+`;
+const H5 = styled.h5`
+  ${headerStyles};
+  font-size: 1.333em;
+`;
+const H6 = styled.h6`
+  ${headerStyles};
+`;
+const ThematicBreak = styled.hr`
+  height: 1px;
+  border: 0;
+  background-image: linear-gradient(
+    to right,
+    ${colors.light.bordersSoft},
+    ${colors.light.borders},
+    ${colors.light.bordersSoft}
+  );
+`;
+const Blockquote = styled.blockquote`
+  border-left: 0.25em solid ${colors.light.cards};
+  color: ${colors.light.textSoft};
+  background-color: ${colors.light.cardsSoft};
+  padding: 0.25rem 1rem;
+  margin: 0;
+  margin-bottom: ${customProperties.spacingM};
+
+  > :first-child {
+    margin-top: 0;
+  }
+  > :last-child {
+    margin-bottom: 0;
+  }
+`;
+const Ul = styled.ul`
+  margin: 0;
+  padding-left: 2em;
+  > * + * {
+    margin-top: 0.25em;
+  }
+`;
+const Ol = styled.ol`
+  margin: 0;
+  padding-left: 2em;
+  > * + * {
+    margin-top: 0.25em;
+  }
+`;
 const Li = styled.li``;
-const Table = styled.table``;
-const TableRow = styled.tr``;
-const TableCell = styled.td``;
-const TableHeader = styled.th``;
-const Pre = styled.pre``;
-const Code = styled.code``;
-const InlineCode = styled.code``;
+const Table = styled.table`
+  border-collapse: collapse;
+  font-size: 0.9rem;
+  tbody {
+    border-top: 4px solid ${colors.light.cards};
+  }
+
+  @media screen and (max-width: 40em) {
+    display: block;
+    tbody {
+      display: block;
+    }
+    thead {
+      display: block;
+    }
+    thead tr {
+      position: absolute;
+      top: -9999px;
+      left: -9999px;
+    }
+    ${props => {
+      const tableHeaders = React.Children.toArray(props.children).find(
+        elem => elem.props.mdxType === 'thead'
+      );
+      const rowHeaders = tableHeaders.props.children;
+      return React.Children.toArray(rowHeaders.props.children).reduce(
+        (styles, elem, index) => `
+        ${styles}
+        td:nth-of-type(${index + 1}):before { content: "${
```

Comment:
On small screens the table becomes a block list (see https://flaviocopes.com/css-responsive-table/).

This logic here is to extract the table headers and inject them into each row cell.

![2019-04-29 14 27 26](https://user-images.githubusercontent.com/1110551/56896084-19f51000-6a8b-11e9-8ce4-8238aefb8f89.gif)